### ISSUE TITLE
Add entry to read-only collection: No error message

### ISSUE THREAD

**Issue by SebastianS90** [dani-garcia/bitwarden_rs#457]
When I have read-only access to a collection and attempt to edit an entry, then I get an error message on save (which is fine).
But when I add an entry then it reports that it has been saved successfully, even in the log it says `Outcome: Success` for the `/api/ciphers/create` query. After reloading the vault, the entry disappears.

Server: `mprasil/bitwarden:1.8.0-alpine`

PS: Thank you so much for this amazing server implementation!

**Comment by SebastianS90** [comment]
The new entries are saved, but not in any collection and therefore the user cannot see it. You might still want to fail the save action in that case (user has read-only access) to avoid confusion.

**Comment by janost** [comment]
I'm not able to reproduce this issue.

As a user who has RO access to the collection, attempting to add an item into the collection results in these log entries:
```
[2019-04-14 22:34:37][rocket::rocket][INFO] POST /api/ciphers/create application/json; charset=utf-8:
[2019-04-14 22:34:37][_][INFO] Matched: POST /api/ciphers/create (post_ciphers_create)
[2019-04-14 22:34:37][bitwarden_rs::error][ERROR] No rights to modify the collection. No rights to modify the collection
[2019-04-14 22:34:37][_][INFO] Outcome: Success
[2019-04-14 22:34:37][_][INFO] Response succeeded.
```

Can you please try the latest master build and if the problem still persists, can you please post reproduction steps?

**Comment by SebastianS90** [comment]
I cannot reproduce it with master anymore. So it probably has been fixed somewhere between the `1.8.0` release and current master.
